src/prefect/tasks/files/operations.py

```diff
@@ -219,4 +219,4 @@ def run(
         globpath = Path(path).joinpath(pattern)
         self.logger.debug(f"Glob path: {globpath}")
 
-        return list(path.glob(pattern))
+        return list(globpath.glob(pattern))
```

Comment:
Using `globpath` will fail in most situations since the pattern has already been joined with the path to search. Here's a suggestion for code that works with both `Path` and `str` types:
```suggestion
        return list(Path(path).glob(pattern))
```

Reply:
Ah, I didn't see that.

This actually nicely fits to a question I have (still learning the library). Why doesn't this work? 

```python
globpath = Path(f"{path}/{pattern}")
```

In my case I get things like:
```
'<Parameter: Path to the top level of the experiment tree>/outdata/fesom/*.nc'
```